lib/promise.js

```diff
@@ -1,33 +1,43 @@
 'use strict';
 
 Promise.each =  async function(arr, fn) {
-   for(const item of arr) await fn(item);
+    for(const item of arr) await fn(item);
 }
 
-Promise.prototype.return = func => {
-  return Promise.resolve(func);
+Promise.prototype.return = function(value) {
+  return this.then(function(){ return value });
 }
 
 Promise.prototype.tap = function(fn) {
-    return this.then(function(value){
-         fn(value);
+    return this.then(async function(value){
+         await fn(value);
          return value;
     });
 };
 
-Promise.try = func => {
-  return new Promise((resolve, reject) => {
-  		resolve(func());
-  	});
-  }
+Promise.try = function(func) {
+    return new Promise(function(resolve, reject) {
+        var val = func();
+        return !val ? reject(val) : resolve(val);
```

Comment:
Do we want to alter the interface here? AFAIK the try method is only really used for catching synchronous errors and does allow for falsey return values.

Reply:
@Phara0h I agree with @rerolled here. This implementation isn't even close to what bluebird does: http://bluebirdjs.com/docs/api/promise.try.html
Are you sure that's safe to do?

Reply:
Yes it works use there test code to try it out

function getUserById(id) {
    return Promise.try(function() {
        if (typeof id !== "number") {
            throw new Error("id must be a number");
        }
        return "yolo"
    });
}

getUserById('sqf').then(function(a){
  console.log(a);
}).catch(e=>console.log('error'))


This will output 'error'